src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -259,6 +259,51 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
             return document.Document.WithSyntaxRoot(newRoot);
         }
 
+        private int GetFirstStatementAffectedIndex(SyntaxNode innermostCommonBlock, ISet<ExpressionSyntax> matches, int firstStatementAffectedIndex)
+        {
+            // If a local function is involved, we have to make sure the new declaration is placed:
+            //     1. Before all calls to local functions that use the variable.
+            //     2. Before the local function(s) themselves.
+            //     3. Before all matches, i.e. places in the code where the new declaration will replace existing code.
+            // Cases (2) and (3) are already covered by the 'firstStatementAffectedIndex' parameter. Thus, all we have to do is ensure we consider (1) when
+            // determining where to place our new declaration.
+
+            // Find all the local functions within the scope that will use the new declaration.
+            var localFunctions = innermostCommonBlock.DescendantNodes().Where(node => node.IsKind(SyntaxKind.LocalFunctionStatement) && matches.Any(match => match.Span.OverlapsWith(node.Span)));
+
+            if (localFunctions.IsEmpty())
+            {
+                return firstStatementAffectedIndex;
+            }
+
+            var localFunctionIdentifiers = localFunctions.Select(node => ((LocalFunctionStatementSyntax)node).Identifier.ValueText);
+
+            // Find all calls to the applicable local functions within the scope.
+            var localFunctionCalls = innermostCommonBlock.DescendantNodes().Where(node => node is InvocationExpressionSyntax invocationExpression &&
+                                                                                  invocationExpression.Expression.GetRightmostName() != null &&
+                                                                                  !invocationExpression.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression) &&
+                                                                                  localFunctionIdentifiers.Contains(invocationExpression.Expression.GetRightmostName().Identifier.ValueText));
+
+            if (localFunctionCalls.IsEmpty())
+            {
+                return firstStatementAffectedIndex;
+            }
+
+            // Find which call is the earliest.
+            var earliestLocalFunctionCall = localFunctionCalls.Min(node => node.SpanStart);
+
+            var statementsInBlock = GetStatements(innermostCommonBlock);
+
+            // Check if our earliest call is before all local function declarations and all matches, and if so, place our new declaration there.
+            var statement = statementsInBlock.Where(s => s.Span.Contains(earliestLocalFunctionCall)).Single();
```

Comment:
Instead of `.Where()` you could use `.IndexOf()` to find the index of the earliest call. Since the statement itself isn't being used here.